Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -747,7 +801,7 @@ public class DependencyResolver<
         //
         // FIXME: We should have a test for this, probably by adding some kind
         // of statistics on the number of backtracks.
-        for constraint in constraints {
+        for constraint in constraints where constraint.requirement != .unversioned {
```

Comment:
Is this hunk actually needed? Semantically, I'm not sure there is a problem with merging an unversioned constraint, it should just not do anything other than validate the assignment is to "unversioned", right?